app/celery/tasks.py

```diff
@@ -313,8 +316,8 @@ def send_email(service_id, notification_id, from_address, encrypted_notification
         except EmailClientException as e:
             current_app.logger.exception(e)
             notification_db_object.status = 'failed'
+            dao_update_notification(notification_db_object)
```

Comment:
That update statement is to update the status when there is an exception. If it is outside the exception block we are always updating the notification but we do not need it, as we update the notification after the notification is successfully sent to client provider.